### PR TITLE
[VAN-1191] Move summary file generation later so that function arena size is available

### PR DESCRIPTION
**Pull request by tim-hoffman**

### CODE REVIEW

File: circuit_passes/src/passes/loop_unroll/mod.rs [hunk 1]
```diff
@@ -506,6 +503,7 @@ mod test {
                 number_of_components: 0,
             })],
             functions: vec![],
+            ..Default::default()
```
Comment (0xddom) [inline circuit_passes/src/passes/loop_unroll/mod.rs:506]: Uuuh, I like this syntax. I didn't knew it

**Review by 0xddom — APPROVED** [review]
lgtm. no backdoors, right? 😉

**Comment by tim-hoffman** [comment]
> lgtm. no backdoors, right? 😉

I need to make a couple years of legit contributions first so you'll never suspect it.

**Review by iangneal — APPROVED** [review]
Sweet!

**Pull request merged**